controlplane/src/core/repositories/analytics/MetricsDashboardRepository.ts

```diff
@@ -178,215 +127,392 @@ export class MetricsDashboardRepository {
    * Get request rate metrics
    */
   public async getRequestRateMetrics({ range = 24, endDate = getEndDate(), params }: GetMetricsProps) {
-    const prange = getPRange(range);
-
-    // get median RPM requests in last [range]h
-    const rate = this.client.query({
-      query: `quantile(0.5, sum(rate(cosmo_router_http_requests_total{${this.getQueryLabels(
-        params,
-      )}}[${range}h])) * 60)`,
-    });
-
-    const prevRate = this.client.query({
-      query: `quantile(0.5, sum(rate(cosmo_router_http_requests_total{${this.getQueryLabels(
-        params,
-      )}}[${range}h] offset ${range}h)) * 60)`,
-    });
+    const granule = getGranularity(range);
+
+    const [start, end] = getDateRange(endDate, range);
+    const [prevStart, prevEnd] = getDateRange(endDate, range, range);
+
+    const multiplier = range * 60;
+
+    // round(quantileDeterministic(0.5)(rate, 1), 4)
+    // get median request rate in last [range]h
+    const queryRate = (start: number, end: number) => {
+      return this.chClient.queryPromise<{ medianRate: number | null }>(`
+      SELECT round(sum(rate) / ${multiplier}, 4) AS medianRate FROM (
+        SELECT
+          toDateTime('${start}') AS startDate,
+          toDateTime('${end}') AS endDate,
+          sum(TotalRequests) AS rate
+        FROM operation_request_metrics_5_30_mv
+        WHERE Timestamp >= startDate AND Timestamp <= endDate
+          AND OrganizationID = '${params.organizationId}'
+          AND FederatedGraphID = '${params.graphId}'
+        GROUP BY Timestamp, OperationName 
+      )
+    `);
+    };
+    const medianRate = queryRate(start, end);
+    const prevMedianRate = queryRate(prevStart, prevEnd);
 
     // get top 5 operations in last [range] hours
-    const top5 = this.client.query({
-      query: `topk(5, sum by (wg_operation_name) (rate(cosmo_router_http_requests_total{${this.getQueryLabels(
-        params,
-      )}}[${range}h])) * 60)`,
-    });
-
-    // get requests in last [range] hours in series of [step]
-    const series = this.client.queryRange({
-      query: `sum(rate(cosmo_router_http_requests_total{${this.getQueryLabels(params)}}[${prange}]) * 60)`,
-      ...this.getRangeQueryProps({
-        range,
-        endDate,
-      }),
-    });
+    const top5 = this.chClient.queryPromise<{ name: string; value: string }>(`
+      WITH
+        toDateTime('${start}') AS startDate,
+        toDateTime('${end}') AS endDate
+      SELECT name, round(sum(rate) / ${multiplier}, 4) AS value FROM (
+        SELECT
+          Timestamp as timestamp,
+        OperationName as name,
+          sum(TotalRequests) as rate
+        FROM operation_request_metrics_5_30_mv
+        WHERE Timestamp >= startDate AND Timestamp <= endDate
+          AND OrganizationID = '${params.organizationId}'
+          AND FederatedGraphID = '${params.graphId}'
+        GROUP BY Timestamp, OperationName 
+      ) GROUP BY name ORDER BY value DESC LIMIT 5
```

Comment:
Let's show the top 10.

Reply:
Sticking with 5 in this view, more will occupy too much space for other charts that will be added here.

I'm adding a `view all` link under each metric to the traces page grouped by operation name and sorted by requests/latency/error rate